pkg/cluster/deploybaseresources.go

```diff
@@ -254,65 +256,84 @@ func (m *manager) subnetsWithServiceEndpoint(ctx context.Context, serviceEndpoin
 	return subnets, nil
 }
 
-func (m *manager) attachNSGs(ctx context.Context) (bool, error) {
+func (m *manager) attachNSGs(ctx context.Context) error {
+	return m._attachNSGs(ctx, 3*time.Minute)
+}
+
+func (m *manager) _attachNSGs(ctx context.Context, timeout time.Duration) error {
```

Comment:
Curious as to why you've avoided changing the function signature of attachNSGs when it comes to the args but changed the return values anyway? Also I think the reasoning for 3m should be documented in a comment or if it's a standard value put somewhere else as a const